pkg/codegen/docs/gen.go

```diff
@@ -2026,12 +1957,10 @@ func (dctx *docGenContext) getMod(
 		}
 
 		if modName != "" && codegen.PkgEquals(tokenPkg, pkg) {
-			parentName := path.Dir(modName)
 			// If the parent name is blank, it means this is the package-level.
-			if parentName == "." || parentName == "" {
+			parentName := ":" + modName + ":"
```

Comment:
This is causing an infinite loop on pulumi-google-native

```
(*docGenContext) getMod(
  token = "google-native:accesscontextmanager/v1:AccessLevel"
  tokenPkg = packageDefRef{pkg: Name: "google-native"}
  pkg = packageDefRef{pkg: Name: "google-native" }
  tool = "Pulumi Docs Generator"
  add = true
)
```

subsequently it loops with parentName = ":accesscontextmanager/v1:"